apps/openmw/mwrender/shadow.cpp

```diff
@@ -0,0 +1,111 @@
+#include "shadow.hpp"
+
+#include <osgShadow/ShadowSettings>
+#include <osgShadow/ViewDependentShadowMap>
+
+#include <components/settings/settings.hpp>
+
+using namespace osgShadow;
+
+namespace MWRender
+{
+
+Shader::ShaderManager::DefineMap ShadowManager::getShadowDefines()
+{
+    Shader::ShaderManager::DefineMap definesWithShadows;
+    definesWithShadows["shadows_enabled"] = "1";
+    definesWithShadows["texture_offset"] = std::to_string(mOutdoorShadowTechnique->getTextureOffset());
+    definesWithShadows["num_pssm_texture"] = std::to_string(mOutdoorShadowTechnique->getSplitCount());
+    definesWithShadows["pssm_texture_size"] = std::to_string(mOutdoorShadowTechnique->getTextureResolution());
+    return definesWithShadows;
+}
+
+Shader::ShaderManager::DefineMap getShadowsDisabledDefines()
+{
+    Shader::ShaderManager::DefineMap definesWithShadows;
+    definesWithShadows["shadows_enabled"] = "0";
```

Comment:
This won't work - the preprocessor fails on undefined references, even if they're `#if`ed out.

Reply:
thx i forgot them